### PR TITLE
RavenDB-7650 WIP - for code review

### PR DESCRIPTION
**Pull request by myarichuk**
changes:
* move request latency too long to performance hints
* change how the "alerts" are stored to make them aggregatable in a similar way to page size perf hints

(I'd like feedback on the way I am aggregating the performance hints)

### CODE REVIEW

File: src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs [hunk 1]
```diff
@@ -1206,7 +1206,7 @@ public RunningBackup OnGoingBackup(long taskId)
             using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
             using (context.OpenReadTransaction())
             {
-                foreach (var collection in _database.DocumentsStorage.GetCollections(context).ToList())
+                foreach (var collection in _database.DocumentsStorage.GetCollections(context))
                 {
```
Comment (myarichuk) [inline src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs:1210]: related to RavenDB-8369 code review -> no need for ToList() here

File: src/Raven.Server/Documents/Handlers/QueriesHandler.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ public class QueriesHandler : DatabaseRequestHandler
         [RavenAction("/databases/*/queries", "POST", AuthorizationStatus.ValidUser)]
         public async Task Post()
         {
-            using (TrackRequestTime())
+            using (TrackRequestTime("POST /databases/*/queries"))
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/QueriesHandler.cs:32]: Don't we need the actual query here as well?

File: src/Raven.Server/NotificationCenter/Notifications/Details/RequestLatencyDetail.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                djv[key] = list;
+            }
+
+            return djv;
+        }
+    }
+
+    public struct RequestLatencyInfo : IDynamicJsonValueConvertible
+    {
+        public readonly string QueryString;
```
Comment (ayende) [inline src/Raven.Server/NotificationCenter/Notifications/Details/RequestLatencyDetail.cs:50]: This is no longer a valid way for us to pass queries, we need to have the actual query / params / stuff that can help the user figure this out here.

File: src/Raven.Server/NotificationCenter/RequestLatency.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        
+        public RequestLatency(NotificationCenter notificationCenter, NotificationsStorage notificationsStorage)
+        {
+            _notificationCenter = notificationCenter;
+            _notificationsStorage = notificationsStorage;
+        }
+
+        public void AddHint(string queryString, long duration, string action)
+        {
+            var requestLatencyPerformanceHint = GetOrCreatePerformanceLatencies(out var details);
```
Comment (ayende) [inline src/Raven.Server/NotificationCenter/RequestLatency.cs:28]: This make me nervous on the ground of multi threaded behavior.
How does this behave in this regard?

File: src/Raven.Server/NotificationCenter/Notifications/Details/RequestLatencyDetail.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.NotificationCenter.Notifications.Details
+{
+    public class RequestLatencyDetail : INotificationDetails
+    {
+        public readonly Dictionary<string, Queue<RequestLatencyInfo>> RequestLatencies = new Dictionary<string, Queue<RequestLatencyInfo>>();
```
Comment (ayende) [inline src/Raven.Server/NotificationCenter/Notifications/Details/RequestLatencyDetail.cs:10]: What limits the size of this?

**Review by ayende — CHANGES_REQUESTED** [review]

**Pull request merged**